src/utils/stringifyUrlObject.ts

```diff
@@ -0,0 +1,70 @@
+import type { ParsedUrlQueryInput } from 'querystring';
+import type { UrlObject } from 'url';
+import queryString from 'query-string';
+
+const isDynamic = (part: string) => part.startsWith('[') && part.endsWith(']');
+const parseDynamicName = (part: string) => part.slice(1, -1);
+const queryToString = (
+  query: Record<string, unknown> | string | undefined,
+  usedQueryParts: string[]
+) => {
+  if (!query) {
+    return;
+  }
+
+  if (typeof query === 'string') {
+    return query;
+  }
+
+  const queryWithoutUsed = Object.entries(query).reduce((acc, [key, value]) => {
+    if (usedQueryParts.includes(key)) {
+      return acc;
+    }
+
+    return {
+      ...acc,
+      [key]: value,
+    };
+  }, {});
```

Comment:
Isn't this just a filter by `key` not in `usedQueryParts`? The reduce should be removed to add clarity to the code if this can be done without reduce, e.g.
```typescript
const queryWithoutUsed = Object.fromEntries(
    Object.entries(query).filter(([key]) => !usedQueryParts.includes(key))
  );
```

Reply:
And if you want to make the runtime complexity less O(n*m) → O(n):
```typescript
const usedQueryPartsSet = new Set(usedQueryParts);
  const queryWithoutUsed = Object.fromEntries(
    Object.entries(query).filter(([key]) => !usedQueryPartsSet.has(key))
  );
```
by using Set lookup inside the filter, string[] lookup is O(n), Set lookup is O(1).

Reply:
Or alternatively using lodash without the speedup for conciseness:
```typescript
import { omit } from 'lodash';
...
const queryWithoutUsed = omit(query, usedQueryParts);
```

Reply:
this whole util is from monorepo, maybe we can refactor it but then in all places

Reply:
let's refactor later

Reply:
lets fix in separate PR